File: 0693-binary-number-with-alternating-bits/0693-binary-number-with-alternating-bits.py
class Solution:
    def hasAlternatingBits(self, n: int) -> bool:
        prev = -1
        while(n):
            curr = n & 1
            if(prev != -1 and prev == curr):
                return False
            
            prev = curr
            n >>= 1
        
        return True
        

File: 0693-binary-number-with-alternating-bits/0693-binary-number-with-alternating-bits.go
func hasAlternatingBits(n int) bool {
    prev := (n & 1) ^ 1
    for n > 0 {
        if prev == (n & 1) {
            return false
        }
        prev = (n & 1)
        n >>= 1
    }
    return true
}